mlkem/kem.c

```diff
@@ -127,7 +178,7 @@ int crypto_kem_enc(uint8_t *ct, uint8_t *ss, const uint8_t *pk) {
  *              - const uint8_t *sk: pointer to input private key
  *                (an already allocated array of MLKEM_SECRETKEYBYTES bytes)
  *
- * Returns 0.
+ * Returns 0 on success, and -1 if the secret key hash check fails
```

Comment:
Seeing that this is the top-level API, add reference to the relevant section of the FIPS203 spec?